src/Endpoints/class-analytics-posts-api-proxy.php

```diff
@@ -53,13 +53,15 @@ protected function generate_data( $response ): array {
 		return array_map(
 			static function( stdClass $item ) use ( $date_format, $site_id ) {
 				return (object) array(
-					'author'  => $item->author,
-					'date'    => wp_date( $date_format, strtotime( $item->pub_date ) ),
-					'id'      => $item->url,
-					'dashUrl' => Parsely::get_dash_url( $site_id, $item->url ),
-					'title'   => $item->title,
-					'url'     => $item->url,
-					'views'   => $item->metrics->views,
+					'author'         => $item->author,
+					'dashUrl'        => Parsely::get_dash_url( $site_id, $item->url ),
+					'date'           => wp_date( $date_format, strtotime( $item->pub_date ) ),
+					'id'             => $item->url,
+					'postId'         => url_to_postid( $item->url ), // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.url_to_postid_url_to_postid
```

Comment:
1. The `id` key has been used for React at some places and it is not a new field. `postId` can return 0 if the post is not found, so it is not suitable for an ID, since we can have multiple 0s. For now, URLs are still better for uniqueness.
3. I have no issue with it switching permalink configurations, as it keeps working.

Reply:
Can we please have point 1 as comment somewhere?

Reply:
I've added a couple of comments here in the hope that it will be clearer. This section might change as the API gets updated and as we start sending Post IDs in metadata.